### PR TITLE
add support for GoLang 1.16

### PR DESCRIPTION
**Pull request by dgrove-oss**

### CODE REVIEW

**Review by rabbah — COMMENTED** [review]
lgtm but think @sciabarra should have a look.

**Comment by dgrove-oss** [comment]
@sciabarra  -- I was planning on getting go 1.16 support in, then updating the CHANGELOG and kicking off the release process next week.  That ok with you, or do you have mor changes to get in before we release?

**Comment by sciabarracom** [comment]
I added support for symlinks because it was needed to support virtualenvs built by the image itself. I do not plan other changes. I am only worried if my change broke "something". Before, permissions were set to a default value (0644) now they are set by the value in the zip file, so if someone has an action with broken permissions in the zip... the action could break. Maybe. Except in the runtime the proxy is run as a root so probably it does not matter.

**Comment by sciabarracom** [comment]
Let me review the changes and test if and I will confirm the PR.

File: common/gobuild.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def build(parent, source_dir, target):
     if os.path.isdir("%s/main" % source_dir):
         source_dir += "/main"
     p = subprocess.Popen(
-        ["go", "build", "-i", "-ldflags=-s -w",  "-o", target],
```
Comment (dgrove-oss) [inline common/gobuild.py:65]: the `-i`  flag has been deprecated for a while; `go build 1.16` takes it a step further by removing it entirely

File: docs/ACTION.md [hunk 2]
```diff
@@ -127,7 +127,7 @@ done
 
 Note here we are just interested in the payload, but in general you may also want to retrieve other fields.
 
-Note the `actionloop` image will accept any source and will try to run it (if it is possible), while the `action-golang-v1.15`  will instead try to compile the sources assuming it is Golang instead.
+Note the `actionloop` image will accept any source and will try to run it (if it is possible), while the `action-golang-v1.N`  will instead try to compile the sources assuming it is Golang instead.
```
Comment (dgrove-oss) [inline docs/ACTION.md:130]: got tired of bumping this number (and similar in docs).   I think `v1.N` is clear, but I'm also ok with backing this bit of this change out if someone things it is unclear.

Comment (rabbah) [reply docs/ACTION.md:130]: I thought this was fine/obvious.

**Comment by sciabarra** [comment]
Some time ago I found an issue building with certain go libraries, will take the time to investigate this issue.

**Comment by sciabarracom** [comment]
Uhm 
```
java.lang.IllegalStateException: You are using version 2.6.12 of Akka, but it appears you (perhaps indirectly) also depend on older versions of related artifacts. You can solve this by adding an explicit dependency on version 2.6.12 of the [akka-cluster-metrics, akka-slf4j, akka-distributed-data, akka-testkit, akka-stream-testkit, akka-cluster-tools] artifacts to your project. See also: https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed
```
Were you able to run the tests?

**Comment by dgrove-oss** [comment]
Yes.  Tests ran just fine (both for me and in travis).  You may need to update your local cloud of core `openwhisk` and do a clean & reinstall of it first.

**Comment by mrutkows** [comment]
@sciabarracom 

> You may need to update your local clone of core `openwhisk` and do a clean & reinstall of it first.

You may want to wipe your GOPATH pkg folder...

**Comment by dgrove-oss** [comment]
rebased to reconcile with #151.   Can we make progress on this soon?  I would like to get the VOTE going for an openwhisk-runtime-go release this week so I can then update all the action loop based runtimes and get them released next week.  If we want to get an openwhisk core release out in August we need to keep things moving.

**Review by sciabarracom — APPROVED** [review]
Run the tests and all the examples against it and worked fine.
Also it works with complex modules like aws.
Check PR #115 to include also this example. 
Then releasing is fine for me.

**Pull request merged**